libs/server-sdk/src/data_components/serialization_adapters/json_deserializer.cpp

```diff
@@ -10,8 +10,8 @@
 namespace launchdarkly::server_side::data_components {
 
 JsonDeserializer::JsonDeserializer(
-    data_interfaces::ISerializedDataReader& reader)
-    : flag_kind_(), segment_kind_(), reader_(reader) {}
+    std::shared_ptr<data_interfaces::ISerializedDataReader> reader)
+    : flag_kind_(), segment_kind_(), reader_(std::move(reader)) {}
```

Comment:
`ISerializedDataReader` is shared ownership because it's constructed outside the SDK. 

Previously, the `JsonDeserializer` held a reference to it and the Lazy Load system held the shared_ptr - but it's simpler to just have the deserializer own it. That way, the Lazy Load system only needs to own one thing instead of two.